src/ripple/protocol/impl/SField.cpp

```diff
@@ -139,6 +140,9 @@ CONSTRUCT_TYPED_SFIELD(sfSignerListID,          "SignerListID",         UINT32,
 CONSTRUCT_TYPED_SFIELD(sfSettleDelay,           "SettleDelay",          UINT32,    39);
 CONSTRUCT_TYPED_SFIELD(sfTicketCount,           "TicketCount",          UINT32,    40);
 CONSTRUCT_TYPED_SFIELD(sfTicketSequence,        "TicketSequence",       UINT32,    41);
+CONSTRUCT_TYPED_SFIELD(sfTokenTaxon,            "TokenTaxon",           UINT32,    42);
```

Comment:
I wouldn't use "token" for nft. That's too generic a term for what NFTs are. I'd rename this `NFTTaxon`.

We should have consistent nomenclature for the new NFT functionality. I'd use `nft` everywhere. It's short and won't be confused with other functionality. Consider renaming all the `NFToken` files and functions to `nft` and all the `token` functions that are nft specific to `nft`.

Consider renaming the other nft functions - like `minter` and `OfferNode` so they either live in an `nft` namespace or have an `nft` prefix. `minter` is less confusing than `OfferNode` or `transferFee`, but I'd change them all.

Reply:
First, @seelabs, I really appreciate having your eyes on the pull request.  You've spotted a bunch of good things to change.  Thanks.

I'm fine with doing the renames, but I need confirmation from @nikb that he's okay with it before I proceed.

I'm thinking that I'll split the update into two commits.  One where I fix the bugs and non-naming improvements that are identified.  Then I'd leave all the renames for a second commit.  That will both improve the readability of the commits and allow me to get going on the non-rename portion of the changes right away.  It may also take a few iterations before we get the renames all settled to everyone's satisfaction.  Does that work for you?

Reply:
Given the recent work in the documentation to change the term `IOU` to `Token` I have agree with you.  Unqualified `token` is a poor synonym for `NFToken`.  Thanks very much for pointing that out.

However I think there's too much inertia in the spelling `NFToken` in the API to switch that to just `NFT`.  `NFToken` is everywhere throughout the spec and folks have been implementing against the transactions named `TFToken*`.

But I agree with you that the `SField`s that use just a `Token` designation are too vague given the current state of the documentation.  I'm going to suggest that we rename the `SField`s that use just `Token` to use `NFToken`.  So we'd get...

- `sfNFTokenTaxon`
- `sfMintedNFTokens`
- `sfBurnedNFTokens`

and the like.

There are also a couple of `SField`s that fully spell out `NonFungibleToken`.  My inclination is to rename those to `NFToken` as well.  That, also, will increase the regularity of the naming.

Thoughts?